src/Workspaces/Core/Portable/Workspace/Solution/DocumentId.cs

```diff
@@ -39,28 +42,20 @@ private DocumentId(ProjectId projectId, Guid guid, string? debugName)
         /// <param name="projectId">The project id this document id is relative to.</param>
         /// <param name="debugName">An optional name to make this id easier to recognize while debugging.</param>
         public static DocumentId CreateNewId(ProjectId projectId, string? debugName = null)
-        {
-            if (projectId == null)
-            {
-                throw new ArgumentNullException(nameof(projectId));
-            }
-
-            return new DocumentId(projectId, Guid.NewGuid(), debugName);
-        }
+            => CreateFromSerialized(projectId, Guid.NewGuid(), isSourceGenerated: false, debugName);
 
         public static DocumentId CreateFromSerialized(ProjectId projectId, Guid id, string? debugName = null)
+            => CreateFromSerialized(projectId, id, isSourceGenerated: true, debugName);
```

Comment:
>true [](http://example.com/codeflow?start=70&length=4)

I'm not quite clear on why this defaults to true.

Reply:
ack!  mistype.